R/gitstats_functions.R

```diff
@@ -111,25 +114,50 @@ add_team_member <- function(gitstats_obj,
 #' @description  List all repositories for an organization, a team or by a
 #'   keyword.
 #' @param gitstats_obj A GitStats object.
+#' @param add_contributors A logical parameter to decide whether to add
+#'   information about repositories' contributors to the repositories output
+#'   (table) when pulling them by organizations (`orgs`) or `phrase`. By default
+#'   it is set to `FALSE` which makes function run faster as, in the case of
+#'   `orgs` search parameter, it reaches only `GraphQL` endpoint with a query on
+#'   repositories, and in the case of `phrase` search parameter it reaches only
+#'   `repositories REST API` endpoint. However, the pitfall is that the result
+#'   does not convey information on contributors. \cr\cr When set to `TRUE`,
+#'   `GitStats` iterates additionally over pulled repositories and reaches to
+#'   the `contributors APIs`, which makes it slower, but gives additional
+#'   information. The same may be achieved with running separately function
+#'   `get_repos_contributors()` on the `GitStats` object with the `repositories`
+#'   output. \cr\cr When pulling repositories by \bold{`team`} the parameter
+#'   always turns to `TRUE` and pulls information on `contributors`.
 #' @return A `GitStats` class object with updated `$repos` field.
 #' @examples
 #' \dontrun{
 #' my_gitstats <- create_gitstats() %>%
-#'   set_connection(
+#'   set_host(
 #'     api_url = "https://api.github.com",
 #'     token = Sys.getenv("GITHUB_PAT"),
 #'     orgs = c("r-world-devs", "openpharma")
 #'   ) %>%
-#'   set_connection(
+#'   set_host(
 #'     api_url = "https://gitlab.com/api/v4",
 #'     token = Sys.getenv("GITLAB_PAT_PUBLIC"),
 #'     orgs = "mbtests"
 #'   ) %>%
 #'   get_repos()
 #' }
 #' @export
-get_repos <- function(gitstats_obj) {
-  gitstats_obj$get_repos()
+get_repos <- function(gitstats_obj, add_contributors = FALSE) {
```

Comment:
I'd rename `get_*` functions to `pull_*` as it seems to me to explain the function logic better.
Same with `setup`, the better and more consistent name would be sth like `set_search`, `set_search_params` or sth like that.

Reply:
Thank you @krystian8207 I created an issue to address it, it will go with this release.